### PR TITLE
feat(portal): add buyer and seller

### PR DESCRIPTION
**Pull request by dorucioclea**
### Pull-Request Checklist

- [x] Code is up-to-date with the `main` branch
- [x] `yarn lint` passes with this change
- [x] `yarn clarinet:check` passes with this change
- [x] `yarn clarinet:tests` passes with this 
- [x] `yarn unit-tests` passes with this
- [x] Run `yarn eol`
- [x] Run `yarn format`
- [x] Run `yarn cleanup`
- [x] This pull request links relevant issues as `Fixes #0000`
- [x] There are new or updated unit tests validating the change
- [x] Documentation has been updated to reflect this change
- [x] The new commits follow conventions outlined in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/)

### CODE REVIEW

File: standalone/portal-api/src/modules/buyer/buyers.controller.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  @Post()
+  async createEntity(
+    @Body() entity: CreateBuyerRequest,
+  ): Promise<GetBuyerResponse> {
+    return await this.buyerService.createEntity(entity);
+  }
+
+  @Post('/:id')
+  async updateEntity(
+    @Param('id') id: string,
```
Comment (Aman-zishan) [inline standalone/portal-api/src/modules/buyer/buyers.controller.ts:27]:
```suggestion
  @PUT('/:id')
  async updateEntity(
    @Param('id') id: string,
```

Is the use of `POST` method here intentional or did you mean to use `PUT`

File: standalone/portal-api/src/modules/relationship/relationship.controller.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    @Param('supplierId') supplierId: string,
+    @Body() entity: CreateRelationshipRequest,
+  ): Promise<GetRelationshipResponse> {
+    return await this.relationshipService.createEntity(entity, buyerId, supplierId);
+  }
+
+  @Post('/:id/:buyerId/:supplierId')
+  async updateEntity(
+    @Param('id') id: string,
+    @Param('buyerId') buyerId: string,
```
Comment (Aman-zishan) [inline standalone/portal-api/src/modules/relationship/relationship.controller.ts:30]:
```suggestion

  @Put('/:id/:buyerId/:supplierId')
  async updateEntity(
    @Param('id') id: string,
    @Param('buyerId') buyerId: string,
```

File: standalone/portal-api/src/modules/supplier/suppliers.controller.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  ): Promise<GetSupplierResponse> {
+    console.log(JSON.stringify(entity, null, 2));
+    return await this.supplierService.createEntity(entity);
+  }
+
+  @Post('/:id')
+  async updateEntity(
+    @Param('id') id: string,
+    @Body() entity: UpdateSupplierRequest,
+  ): Promise<GetSupplierResponse> {
```
Comment (Aman-zishan) [inline standalone/portal-api/src/modules/supplier/suppliers.controller.ts:30]:
```suggestion

  @Put('/:id')
  async updateEntity(
    @Param('id') id: string,
    @Body() entity: UpdateSupplierRequest,
  ): Promise<GetSupplierResponse> {
```

File: standalone/portal-api/src/modules/supplier/dto/request/update-supplier-company-tax-and-revenue.dto.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  
+    @ApiProperty({ example: '1000000000' })
+    @IsNumberString()
+    exportValue?: number;
+  
+
+    @ApiProperty({ example: '1000000000' })
+    @IsNotEmpty()
+    @IsBoolean()
+    audited: boolean;
```
Comment (Aman-zishan) [inline standalone/portal-api/src/modules/supplier/dto/request/update-supplier-company-tax-and-revenue.dto.ts:28]:
```suggestion
    @ApiProperty({ example: 'true' })
    @IsBoolean()
    audited?: boolean;
```

Currently in designs this field is not required in quick application, but i think we may have to iterate over DTO changes to match designs over development period anyway

**Review by Aman-zishan — APPROVED** [review]
@dorucioclea a few suggestions other than that this looks great 🚀

File: standalone/portal-api/src/modules/relationship/relationship.controller.ts (hunk 2 above)
Comment (dorucioclea) [reply standalone/portal-api/src/modules/relationship/relationship.controller.ts:30]: yup, my bad 😄

File: standalone/portal-api/src/modules/supplier/suppliers.controller.ts (hunk 3 above)
Comment (dorucioclea) [reply standalone/portal-api/src/modules/supplier/suppliers.controller.ts:30]: changed it

File: standalone/portal-api/src/modules/supplier/dto/request/update-supplier-company-tax-and-revenue.dto.ts (hunk 4 above)
Comment (dorucioclea) [reply standalone/portal-api/src/modules/supplier/dto/request/update-supplier-company-tax-and-revenue.dto.ts:28]: done

File: standalone/portal-api/src/modules/buyer/buyers.controller.ts (hunk 1 above)
Comment (dorucioclea) [reply standalone/portal-api/src/modules/buyer/buyers.controller.ts:27]: I meant put :) thank you

**Pull request merged**